test/libsolidity/semanticTests/abiEncoderV1/abi_encode_empty_string.sol

```diff
@@ -0,0 +1,34 @@
+contract C {
+    function f1() public returns (bytes memory) {
+        return abi.encode("");
+    }
+    function f2(string calldata msg) public returns (bytes memory) {
+        return abi.encode(msg);
+    }
+    function g1() public returns (bytes memory) {
+        return abi.encodePacked("");
+    }
+    function g2(string calldata msg) public returns (bytes memory) {
+        return abi.encodePacked(msg);
+    }
+    function h1() public returns (bytes memory) {
+        return abi.encodeWithSelector(0x00000001, "");
+    }
+    function h2(string calldata msg) public returns (bytes memory) {
+        return abi.encodeWithSelector(0x00000001, msg);
+    }
+}
+
+// ====
+// ABIEncoderV1Only: true
+// compileViaYul: false
+// ----
+// f1() -> 0x20, 0x60, 0x20, 0, 0
+// f2(string): 0x20, 0 -> 0x20, 0x40, 0x20, 0
+// f2(string): 0x20, 0, 0 -> 0x20, 0x40, 0x20, 0
+// g1() -> 32, 0
```

Comment:
There is no padding in packed as expected, but padding is present in all cases a literal is used.